MapboxNavigation/RouteController.swift

```diff
@@ -110,9 +107,48 @@ extension RouteController: CLLocationManagerDelegate {
             }
         }
         
+        let step = routeProgress.currentLegProgress.currentStepProgress.step
+        if step.maneuverType == .depart && !userIsOnRoute(location) {
+            
+            guard let userSnappedDistanceToClosestCoordinate = closestCoordinate(on: step.coordinates!, to: location.coordinate)?.distance else {
+                return
+            }
+            
+            // Give the user x seconds of moving away from the start of the route before rerouting
+            guard Date().timeIntervalSince(lastTimeStampSpentMovingAwayFromStart) > MaxSecondsSpentTravelingAwayFromStartOfRoute else {
+                lastUserDistanceToStartOfRoute = userSnappedDistanceToClosestCoordinate
+                return
+            }
+            
+            // Don't check `userIsOnRoute` if the user has not moved
+            guard userSnappedDistanceToClosestCoordinate != lastUserDistanceToStartOfRoute else {
+                lastUserDistanceToStartOfRoute = userSnappedDistanceToClosestCoordinate
+                return
+            }
+            
+            lastUserDistanceToStartOfRoute = userSnappedDistanceToClosestCoordinate
+            
+            if userSnappedDistanceToClosestCoordinate > lastUserDistanceToStartOfRoute {
```

Comment:
`userSnappedDistanceToClosestCoordinate` is never greater than `lastUserDistanceToStartOfRoute`, because the line right above sets `lastUserDistanceToStartOfRoute` to be equal to `userSnappedDistanceToClosestCoordinate`.